explainer.md

```diff
@@ -41,9 +40,40 @@ On a system that does not use file extensions but associates files with MIME typ
 
 The user can right click on CSV or SVG files in the operating system's file browser, and choose to open the files with the Grafr web application. (This option would only be presented if Grafr has been [installed](https://w3c.github.io/manifest/#installable-web-applications).)
 
-A new top level browsing context is created, navigating to the file handling url, e.g. grafr.com/file-open
+A LaunchEvent containing a sequence<[FileSystemFileHandle](https://github.com/WICG/writable-files/blob/master/EXPLAINER.md)> is then sent to the service worker, allowing the web application to decide where to open the files (i.e. in a new or existing client).
+
+```js
+  self.addEventListener('file', event => {
+    event.waitUntil(async () => {
+      const allClients = await clients.matchAll();
+      // If there isn't one available, open a window.
+      if (allClients.length === 0) {
+        const client = clients.openWindow('/');
+        client.postMessage(event.files);
+        client.focus();
+        return;
+      }
+
+      const client = allClients[0];
+      client.postMessage(event.files);
+      client.focus();
+    }());
+  });
+```
+
+`event` has the following shape.
 
-A LaunchEvent containing a sequence<[FileSystemFileHandle](https://github.com/WICG/writable-files/blob/master/EXPLAINER.md)> is then sent to the window, allowing the web application to read and update the files.
+```webidl
+  interface FileEvent : ExtendableEvent {
```

Comment:
Why don't we name it LaunchEvent?

Reply:
That's a good call, I'll make the change.